Lib/test/test__xxsubinterpreters.py

```diff
@@ -526,30 +526,25 @@ def test_with_int(self):
         self.assertEqual(int(id), 10)
 
     def test_coerce_id(self):
-        id = interpreters.InterpreterID('10', force=True)
-        self.assertEqual(int(id), 10)
-
-        id = interpreters.InterpreterID(10.0, force=True)
-        self.assertEqual(int(id), 10)
-
         class Int(str):
-            def __init__(self, value):
-                self._value = value
-            def __int__(self):
-                return self._value
+            def __index__(self):
+                return 10
 
-        id = interpreters.InterpreterID(Int(10), force=True)
-        self.assertEqual(int(id), 10)
+        for id in ('10', '1_0', Int()):
+            with self.subTest(id=id):
+                id = interpreters.InterpreterID(id, force=True)
+                self.assertEqual(int(id), 10)
 
     def test_bad_id(self):
-        for id in [-1, 'spam']:
+        for id in (10.0, b'10', []):
             with self.subTest(id):
-                with self.assertRaises(ValueError):
+                with self.assertRaises(TypeError):
                     interpreters.InterpreterID(id)
+        for id in (-1, '-1', 'spam'):
+            with self.assertRaises(ValueError):
+                interpreters.InterpreterID(id)
         with self.assertRaises(OverflowError):
             interpreters.InterpreterID(2**64)
-        with self.assertRaises(TypeError):
-            interpreters.InterpreterID(object())
```

Comment:
It would be nice to have a test still for `object()`.

Reply:
I replaced it with `[]`. But if you prefer, I'll return `object()`.